convert/plan.go

```diff
@@ -48,17 +48,59 @@ type Planner struct {
 	// if it would be converting a TSDB block that spans over more days, to avoid re-downloading that block
 	// for the next plan.
 	maxDays int
+	// Optional relative time window (offsets from now). If both zero, no window filtering.
+	minTimeOffset time.Duration
+	maxTimeOffset time.Duration
 }
 
-func NewPlanner(notAfter time.Time, maxDays int) Planner {
-	return Planner{notAfter: notAfter, maxDays: maxDays}
+type PlannerOption func(*Planner)
+
+// WithTimeWindow sets optional relative time window offsets.
+func WithTimeWindow(minOffset, maxOffset time.Duration) PlannerOption {
+	return func(p *Planner) {
+		p.minTimeOffset = minOffset
+		p.maxTimeOffset = maxOffset
+	}
+}
+
+func NewPlanner(notAfter time.Time, maxDays int, opts ...PlannerOption) Planner {
+	p := Planner{notAfter: notAfter, maxDays: maxDays}
+	for _, opt := range opts {
+		opt(&p)
+	}
+	return p
 }
 
 func (p Planner) Plan(tsdbMetas map[string]metadata.Meta, parquetMetas map[string]schema.Meta) Plan {
```

Comment:
maybe we could also do "Plan(tsdbMetas, parquetMetas, mint, maxt)" with the semantics of only producing blocks between mint and maxt given present parquetmetas and tsdbmetas? wdyt of that api?